security/src/main/java/com/facebook/airlift/security/ReloadableSslContext.java

```diff
@@ -0,0 +1,171 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.facebook.airlift.security;
+
+import com.facebook.airlift.log.Logger;
+import com.facebook.airlift.security.pem.PemReader;
+import com.google.common.hash.HashCode;
+import com.google.common.io.Files;
+import io.airlift.units.Duration;
+
+import javax.net.ssl.KeyManager;
+import javax.net.ssl.KeyManagerFactory;
+import javax.net.ssl.SSLContext;
+import javax.net.ssl.TrustManagerFactory;
+
+import java.io.File;
+import java.io.IOException;
+import java.security.GeneralSecurityException;
+import java.security.KeyStore;
+import java.util.Objects;
+import java.util.Optional;
+import java.util.concurrent.atomic.AtomicReference;
+import java.util.function.Supplier;
+
+import static com.google.common.base.Preconditions.checkState;
+import static com.google.common.hash.Hashing.sha256;
+import static java.util.Objects.requireNonNull;
+
+public final class ReloadableSslContext
+        implements Supplier<SSLContext>
+{
+    private static final Logger log = Logger.get(ReloadableSslContext.class);
+
+    private final FileWatch trustCertificatesFileWatch;
+    private final FileWatch clientCertificatesFileWatch;
+
+    private final AtomicReference<SSLContext> sslContext;
+    private final Duration sslContextRefreshTime;
+    private final Thread sslContextRefreshThread;
```

Comment:
Let's make `sslContextRefreshThread` non final and group it together with the `started` variable. Also let's annotate them with `@GuardedBy("this")` and make the class thread safe, make the `start` and `stop` methods synchronized.